bugtool/cmd/root.go

```diff
@@ -413,6 +428,43 @@ func split(prompt string) (string, []string) {
 	return cmd, args
 }
 
+// hashIPseckeys hashes keys and replaces them with its sha256
+func hashIPseckeys(output string) string {
+	var final []string
+	// splitting the result as an array of string
+	lines := strings.Split(output, "\n")
+	// list of substrings to search for in lines "atuh-trunc" is removed from list
+	// of substrings because line containing "auth-trunc" will be matched by "auth"
+	substrings := [...]string{"auth", "enc", "aead", "comp"}
+	for _, line := range lines {
+		for _, substr := range substrings {
+			// checking if the line contains our substring
+			r, _ := regexp.Compile(substr)
+			matched := r.MatchString(line)
+			if matched {
+				// looking for hex substring which starts with '0x'
+				for i, splits := range strings.Fields(line) {
+					hexFound, _ := regexp.MatchString("0x", splits)
+					if hexFound {
+						splits := strings.Fields(line)
+						pass := strings.Fields(line)[i]
+						h := sha256.New()
+						h.Write([]byte(pass))
+						hash := hex.EncodeToString(h.Sum(nil))
+						splits[i] = "[hash: " + hash + "]"
+						line = strings.Join(splits, " ")
```

Comment:
```suggestion
						line = strings.Join(splits, " ")
						break
```
Or do you expect several hexa strings on a line?

Reply:
No, only one hex for the lines which contain our pattern. Fixed this.